ui/components/app/assets/asset-list/network-filter/network-filter.tsx

```diff
@@ -67,21 +75,34 @@ const NetworkFilter = ({ handleClose }: SortControlProps) => {
       formattedTokensForAllNetworks,
     );
 
-  // TODO: fetch balances across networks
-  // const multiNetworkAccountBalance = useMultichainAccountBalance()
-
   const handleFilter = (chainFilters: Record<string, boolean>) => {
     dispatch(setTokenNetworkFilter(chainFilters));
 
     // TODO Add metrics
     handleClose();
   };
 
+  useEffect(() => {
+    const testnetChains: string[] = TEST_CHAINS;
+    const mainnetChainIds = Object.keys(allNetworks).filter(
+      (chain) => !testnetChains.includes(chain),
+    );
+    setChainsToShow(mainnetChainIds);
+  }, []);
+
+  const allOpts: Record<string, boolean> = {};
+  Object.keys(allNetworks).forEach((chain) => {
+    allOpts[chain] = true;
+  });
```

Comment:
I think that we should refactor this logic into a hook. It's being used in multiple places, and should just use the `getChainIdsToPoll` selector.

I would prefer to address this cleanup in a separate PR, if possible.